rust/connlib/libs/tunnel/src/tun_linux.rs

```diff
@@ -143,12 +145,18 @@ impl IfaceDevice {
             .address()
             .add(self.interface_index, config.ipv4.into(), 32)
             .execute()
-            .await?;
+            .await
+            .or(self
+                .handle
+                .address()
+                .add(self.interface_index, config.ipv6.into(), 128)
+                .execute()
+                .await)?;
```

Comment:
I _think_ this should be `or_else` to delay the execution.

Currently, `adress().add` will always be executed and the result will be discarded if the other `Option` is `Some`.

That is because parameters need to be evaluated first before the actual function is being executed.

Reply:
In that case I think it would be cleaner to execute both functions and then combine the results.